### PR TITLE
fix: reset token selections when  changing chains on /add

### PR DESCRIPTION
**Pull request by just-toby**
## Description

fixes an issue on the `/add` page where token selections from the previous chain persist to the new chain, and are displayed as "UNKNOWN" ...

my fix here is to disallow any "unknown" tokens since they likely don't exist. an "unknown" token is one where we couldn't get valid decimals/name/symbol from the onchain call.

the token state on this page is actually managed via the URL route and is actually hard to change. (i tried using react router to reset the state in many different ways, but couldn't get it working). 
we might want to consider moving away from URL-based state management on this page at some point, but i wanted to submit this simpler fix as a short term solution.

_Linear ticket:_ https://linear.app/uniswap/issue/WEB-3014/if-you-change-network-in-create-position-it-doesnt-clear-input
_Slack thread:_
_Relevant docs:_

### Before

https://github.com/Uniswap/interface/assets/66155195/4593daaa-fb4b-47a9-89b3-bf39f929eef2

### After

https://github.com/Uniswap/interface/assets/66155195/9415c53c-3c7c-4c8f-b183-7db3825c1a2d

### Reproducing the error

1. go to /add/ETH page
2. select tokens
3. change chain
4. notice that the tokens are "unknown" on new chain

### CODE REVIEW

**Comment by JFrankfurt** [comment]
Can you add an e2e test that handles this case?

**Review by JFrankfurt — APPROVED** [review]

File: src/lib/hooks/useCurrency.ts [hunk 1]
```diff
@@ -68,9 +68,12 @@ export function useTokenFromActiveNetwork(tokenAddress: string | undefined): Tok
     // If the token is on another chain, we cannot fetch it on-chain, and it is invalid.
     if (typeof tokenAddress !== 'string' || !isSupportedChain(chainId) || !formattedAddress) return undefined
     if (isLoading || !chainId) return null
+    if (!decimals?.result?.[0] && parsedSymbol === UNKNOWN_TOKEN_SYMBOL && parsedName === UNKNOWN_TOKEN_NAME) {
+      return undefined
+    }
```
[the remark this answers is not in the document]
Comment (JFrankfurt) [reply src/lib/hooks/useCurrency.ts:73]: might be a good time to add a `useCurrency.test.ts`

**Review by mattbspector — APPROVED** [review]

**Review by JFrankfurt — APPROVED** [review]

File: cypress/e2e/add-liquidity.test.ts [hunk 2]
```diff
@@ -16,6 +16,16 @@ describe('Add Liquidity', () => {
     cy.contains('0.05% fee tier')
   })
 
+  it.only('clears the token selection when chain changes', () => {
```
Comment (JFrankfurt) [inline cypress/e2e/add-liquidity.test.ts:19]:
```suggestion
  it('clears the token selection when chain changes', () => {
```

**Review by JFrankfurt — CHANGES_REQUESTED** [review]

File: cypress/e2e/add-liquidity.test.ts (hunk 2 above)
Comment (just-toby) [reply cypress/e2e/add-liquidity.test.ts:19]: dang

**Review by JFrankfurt — APPROVED** [review]

**Pull request merged**